### DISCUSSION TITLE
problema stato entità durante shutdown

### DISCUSSION THREAD

**Discussion by 50494554524F** [indomus/forum#1139] [HUB: Home Assistant]
(
premessa: ho creato degli switch per incapsulare le valvole termostatiche e poi aggiunto questi switch ad un group.
in base allo stato del group richiamo uno bot per accendere o spegnere la caldaia.
fin qui funziona tutto come ci si aspetterebbe...
)

durante lo shutdown il bot entrava in funzione lo stesso.
mi son accorto (suppongo perché non viene registrato lo stato a questo punto) 
che le entità delle valvole cambiano stato su unavailable, quindi
gli switch diventano off, il gruppo va su off e l'automazione per spegnere la caldaia parte. 
al riavvio tornano gli stati corretti ma la corrispettiva automazione per riaccendere non parte, il cambio di stato non c'è.
è un comportamento normale che le entità diventino unavailable?! gli stati non si dovrebbero congelare quando si spegne?!
non trovo niente a riguardo... se si, bisognerà tenere conto anche di questo e bisognerà far funzionare il group al contrario...
grazie

**Comment by 50494554524F** [comment]
che sia un bug o no il mancato salvataggio dello stato off in chiusura, ci ho pensato... invertirò il funzionamento del gruppo, usando on invece che off ci previene da problemi in ogni caso... del gruppo si potrebbe anche fa a meno, lo uso per non dover modificare ogni volta la lista delle entità nei vari file nel caso servisse fare modifiche

**Reply by snakuzzo** [reply]
strana questa cosa.
Io ho appena fatto una prova con un Light Group e con un Group di light

Con un light group, se entrambe le light sono `unavailable` anche il gruppo lo è
Con un group di light invece, se entrambe le light sono `unavailable`, come effettivamente hai fatto notare, il gruppo va in 'off'

Secondo me a questo punto ti conviene basare l'automazione su un template switch o un template sensor, e non su un gruppo

**Reply by 50494554524F** [reply]
non ci avevo pensato, in effetti è la stessa cosa... 

(che poi non è tanto il comportamento del gruppo perché al riavvio la sequenza degli eventi andrebbe al contrario, è che non salva lo stato della caldaia quando la spegne... a tempo libero da investigare)

**Reply by 50494554524F** [reply]
risolto aggiungendo "availability_template" agli switch - così da intercettare quando le valvole vanno su "unavailable" - ed un sensore che è "unavailable" se tutti gli switch sono "unavailable", "off" se tutti gli switch sono "off" ed "on" se almeno uno switch è "on"... in questo modo non ho dovuto nemmeno modificare le automazioni se non group. con sensor. ...

**Reply by snakuzzo** [reply]
perfetto! bravo! :)
